Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts

```diff
@@ -141,4 +212,127 @@ export class JenkinsRestClient {
 
         return defer.promise;
     }
+
+    public GetJobType(): Q.Promise<string> {
+        let defer = Q.defer<string>();
+        let jenkinsJobType: string = tl.getInput("jenkinsJobType", false) || "";
+
+        if (!!jenkinsJobType && jenkinsJobType.trim().length > 0) {
```

Comment:
why do we look at jobType specified as task input?

we should detect jobType from Jenkins and validate that the job/version etc. specified as task input correspond to the job type. if not fail with appropriate error.

Reply:
This is not a user input anymore. If the Jenkins server is reachable, then we would have already detected in our hidden input. To avoid making one more call, i thought we could directly use it, its a not a user input anyway.

Reply:
Fixed it as we discussed.